CONTRIBUTING.md

```diff
@@ -167,7 +167,24 @@ You can add the sign off when creating the git commit via `git commit -s`.
 
 If you want this to be automatic you can set up some aliases:
 
-```
+```bash
 git config --add alias.amend "commit -s --amend"
 git config --add alias.c "commit -s"
 ```
+
+## Fixing DCO
+
+If your PR fails the DCO check, it's necessary to fix the entire commit history in the PR. Best
+practice is to [squash](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html)
+the commit history to a single commit, append the DCO sign-off as described above, and [force
+push](https://git-scm.com/docs/git-push#git-push---force). For example, if you have 2 commits in
+your history:
+
+```bash
+git rebase -i HEAD^^
```

Comment:
I have `upstream/master` for the fork. I think in general, it's hard to give a perfect config agnostic setup, since you might name the remote `foo` and it would be `foo/master`. 

I agree a soft reset is probably more direct. There's two reasons I didn't advice to do it here:

1. It's awesome to learn about the Swiss army knife of interactive rebase if you are new to git, it opens up lots of options later.

2. The flow you describe is adding a `origin/master` rebase, which may be desired, but is technically not needed.

I don't mind if you want to recommend that instead, just giving the motivation I had in mind.

Reply:
Nah it's fine, just throwing it out there.

Reply:
OK, let's see if folks find what I put in simple enough. If not, we can improve in a followup PR.